src/modules/navigator/takeoff.cpp

```diff
@@ -69,6 +69,26 @@ Takeoff::on_active()
 		_navigator->get_mission_result()->finished = true;
 		_navigator->set_mission_result_updated();
 		_navigator->mode_completed(getNavigatorStateId());
+
+		position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
+
+		// set loiter item so position controllers stop doing takeoff logic
+		if (_navigator->get_land_detected()->landed) {
+			_mission_item.nav_cmd = NAV_CMD_IDLE;
+
+		} else {
+			if (pos_sp_triplet->current.valid
+			    && _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
+				setLoiterItemFromCurrentPositionSetpoint(&_mission_item);
+
+			} else {
+				setLoiterItemFromCurrentPosition(&_mission_item);
```

Comment:
In #24454 all of this logic was removed. Is it intended to add back this and below logic? Or should we do the VEHICLE_TYPE_ROTARY_WING check higher up to maintain the changes in #24454 for non-multicopter?

Reply:
Let's play it save now and do the minimal-invasive change compared to main from one week ago. And that means only patch the line here.